subcommand/common/common_test.go

```diff
@@ -26,3 +59,97 @@ func TestValidatePort(t *testing.T) {
 	err = ValidatePort("-test-flag-name", "22")
 	require.EqualError(t, err, "-test-flag-name value of 22 is not in the port range 1024-65535.")
 }
+
+// TestConsulLogin ensures that our implementation of consul login hits `/v1/acl/login`.
+func TestConsulLogin(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+
+	counter := 0
+	consulServer, client, bearerTokenFile, tokenFile := startMockServer(t, "foo", &counter)
+	defer consulServer.Close()
+	err := ConsulLogin(
+		client,
+		bearerTokenFile,
+		testAuthMethod,
+		tokenFile,
+		testPodMeta,
+	)
+	require.NoError(err)
+	require.Equal(counter, 1)
+	// validate that the token file was written to disk
+	data, err := ioutil.ReadFile(tokenFile)
+	require.NoError(err)
+	require.Equal(string(data), "b78d37c7-0ca7-5f4d-99ee-6d9975ce4586")
+}
+
+func TestConsulLogin_EmptyBearerTokenFile(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	counter := 0
+	consulServer, client, bearerTokenFile, tokenFile := startMockServer(t, "", &counter)
+	defer consulServer.Close()
+	err := ConsulLogin(
+		client,
+		bearerTokenFile,
+		testAuthMethod,
+		tokenFile,
+		testPodMeta,
+	)
+	require.Error(err, "unable to read bearerTokenFile")
+}
+
+func TestConsulLogin_NoBearerTokenFile(t *testing.T) {
+	t.Parallel()
+	require := require.New(t)
+	counter := 0
+	consulServer, client, _, tokenFile := startMockServer(t, "foo", &counter)
+	randFileName := fmt.Sprintf("/foo/%d/%d", rand.Int(), rand.Int())
```

Comment:
I picked random file name because I wanted to prevent any chance of accidentally finding "foo" as a valid file on a test. I guess its not likely though.

Reply:
sounds good!